ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,94 @@
+#![allow(unused)]
+use crate::{crypto::hash, ssz::ByteVector};
+use alloy_primitives::{uint, U256};
+use blst::min_pk::PublicKey;
+pub use c_kzg::{Bytes32, Bytes48, Error, KzgSettings};
+use ssz_rs::prelude::*;
+
+pub const BLS_MODULUS: U256 =
+    uint!(52435875175126190479447740508185965837690552500527637822603658699938581184513_U256);
+pub const BYTES_PER_BLOB: usize = 32 * 4096;
+pub const BYTES_PER_FIELD_ELEMENT: usize = 32;
+pub const BYTES_PER_CONTEXT: usize = 10;
+pub const BYTES_PER_COMMITMENT: usize = 48;
+pub const BYTES_PER_PROOF: usize = 48;
+pub const G1_POINT_AT_INFINITY: [u8; 48] = create_g1_point_at_infinity();
+pub const FIELD_ELEMENTS_PER_BLOB: usize = 4096;
+pub const FIAT_SHAMIR_PROTOCOL_DOMAIN: &[u8; 16] = b"FSBLOBVERIFY_V1_";
+
+pub type BLSFieldElement = U256;
+pub type Polynomial = Vec<BLSFieldElement>; // Should this polynomial type be an array?
+
+const fn create_g1_point_at_infinity() -> [u8; 48] {
+    let mut arr: [u8; 48] = [0; 48];
+    arr[0] = 0xc0;
+    arr
+}
+
+/// TODO:  Lean on C-KZG library to implement specs.
+
+pub struct Blob(ByteVector<BYTES_PER_BLOB>);
+
+#[derive(SimpleSerialize, Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct KzgCommitment(ByteVector<BYTES_PER_COMMITMENT>);
+
+#[derive(SimpleSerialize, Default, Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct KzgProof(ByteVector<BYTES_PER_PROOF>);
+
+/// Uses multicalar multiplication to combine trusted setup with blob
+fn blob_to_kzg_commitment(blob: Blob, kzg_settings: &KzgSettings) -> Result<KzgCommitment, Error> {
+    let bytes = blob.0.as_ref();
+    let blob = c_kzg::Blob::from_bytes(bytes).unwrap();
+
+    // Inner: g1_lincomb(bit_reversal_permutation(KZG_SETUP_LAGRANGE), blob_to_polynomial(blob)) -> KzgCommitment
+    let commit = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+
+    let bytes_commit = commit.to_bytes();
+    let bytevector_commit = ByteVector::try_from(bytes_commit.as_ref()).unwrap();
+
+    Ok(KzgCommitment(bytevector_commit))
+}
+
+/// Compute KZG proof at point 'z' for the polynomial represented by 'blob'.
```

Comment:
use backticks if you are referring to code elements

like _`blob`_ instead of _'blob'_